src/figuresClasses.ts

```diff
@@ -1,11 +1,102 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: Color;
+
+  sideA: number;
+
+  sideB: number;
+
+  sideC: number;
+
+  constructor(color: Color, sideA: number, sideB: number, sideC: number) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('any length is <= 0');
+    }
+
+    if (
+      sideA >= sideB + sideC ||
+      sideB >= sideA + sideC ||
+      sideC >= sideA + sideB
+    ) {
+      throw new Error(
+        'the longest side of a triangle is >= than a sum of two others',
+      );
+    }
+    this.shape = 'triangle';
+    this.color = color;
+    this.sideA = sideA;
+    this.sideB = sideB;
+    this.sideC = sideC;
```

Comment:
The constructor parameters don't need to be manually assigned to class properties since TypeScript does this automatically when using access modifiers. Consider adding access modifiers to the parameters and removing the manual assignments.